### PR TITLE
Drop .NET 6 Support

### PR DESCRIPTION
**Pull request by sfmskywalker**
This PR removes support for .NET 6.0 from the project. Updates include:  
- Adjusted project files to target only supported .NET versions.  
- Updated documentation and CI/CD pipelines to reflect the removal of .NET 6.0.  
- Removed any .NET 6.0-specific code or dependencies.  

This change simplifies maintenance and focuses development on currently supported .NET versions.

- - -
This change is [Reviewable](https://reviewable.io/reviews/elsa-workflows/elsa-core/6233)

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -1,234 +1,152 @@
...
-    <PackageVersion Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="7.0.18" />
-    <PackageVersion Include="Oracle.EntityFrameworkCore" Version="7.21.13" />
-    <PackageVersion Include="Polly" Version="7.2.4" />
-    <PackageVersion Include="Pomelo.EntityFrameworkCore.MySql" Version="7.0.0" />
-    <PackageVersion Include="Refit" Version="7.2.22" />
-    <PackageVersion Include="Refit.HttpClientFactory" Version="7.0.0" />
-    <PackageVersion Include="System.Formats.Asn1" Version="8.0.1" />
-    <PackageVersion Include="System.Text.Json" Version="8.0.5" />
-  </ItemGroup>
-  <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
```
Comment (jayachandra21) [inline Directory.Packages.props:152]: Hello @sfmskywalker Removing this causing problem in .Net8.0 Target framework packages

**Review by jayachandra21 — COMMENTED** [review]
In the Directory.packages.props file by removing the <ItemGroup Condition="'$(TargetFramework) is created a problem

File: Directory.Packages.props (hunk 1 above)
Comment (sfmskywalker) [reply Directory.Packages.props:152]: That's odd - the Elsa.Server.Web project seems to build fine when targeting .NET 8.0.

Do you have a repro?

Comment (jayachandra21) [reply Directory.Packages.props:152]: I think as Target Framework 6.0 is removed, particular  to those package section can be removed.
For TargetFramework 8.0 and 9.0, can we still go with validation Condition check as below 
<ItemGroup Condition="'$(TargetFramework)' == 'net8.0'"><PackageVersions>.

Comment (sfmskywalker) [reply Directory.Packages.props:152]: We could, but it’s better if we don’t have to, keeps things simple. I haven’t seen any issues so far with referencing 9.0 packages. Can you show me the issue?

**Pull request merged**